modules/agar/src/main/ts/ephox/agar/assertions/ApproxStructures.ts

```diff
@@ -221,7 +252,21 @@ const anythingStruct: StructAssert = {
   doAssert: Fun.noop
 };
 
-const assertAttrs = (expectedAttrs: Record<string, StringAssert>, actual: SugarElement<Element>) => {
+const assertAttrs = (expectedAttrs: Record<string, StringAssert>, actual: SugarElement<Element>, specificMode: boolean = false) => {
```

Comment:
Generally, try to avoid passing through a boolean that causes branching in the implementation. If you can write the new assertion function by doing a step first, then doing the rest, it would be better to make a separate function, do the additional check first, then call the other one. If it causes some efficiency, you can make a shared function that gets passed the `allDefinedAttrs` or something similar

Reply:
Moved this to separate functions.

The specific checking function uses the key from `allDefinedAttrs`, but the general function retrieves the key and value from `allDefinedAttrs`. The difference is the value is used for string/array assertions.